### PR TITLE
Feature/preserve yaml comments

### LINKED ISSUE #99: Preserve nitric.yaml comments after make
When using the cli to make a new service or another update to a nitric stack yaml file, the comments are lost and ordering changes. The file should have the new additions, but no other changes should occur.

┆Issue is synchronized with this [Jira Story](https://team-1603662928229.atlassian.net/browse/CST-30) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by jyecusch**
Closes: https://github.com/nitrictech/cli/issues/99

┆Issue is synchronized with this [Jira Task](https://team-1603662928229.atlassian.net/browse/CST-14) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/common/src/stack/stack.ts [hunk 1]
```diff
@@ -259,35 +315,34 @@ export class Stack<
 	/**
 	 * Parse a Nitric Stack definition from the given file.
 	 *
-	 * @param file containing the serialized stack definition
+	 * @param file YAML file contents containing the serialized stack definition
 	 * @param parser to parse the serialized file type, defaults to YAML parser.
```
Comment (tjholm) [inline packages/common/src/stack/stack.ts]: remove @param parser

File: packages/common/src/stack/stack.ts [hunk 2]
```diff
@@ -180,6 +229,13 @@ export class Stack<
 		return path.join(STAGING_DIR, this.name);
 	}
 
+	/**
+	 * Return comments data for serialized stack file
+	 */
+	getComments(): comments {
+		return this.comments || [];
+	}
```
Comment (tjholm) [inline packages/common/src/stack/stack.ts]: can get rid of this if comments is defaulted to an empty array in the constructor and made non-optional as a field.

File: packages/common/src/stack/stack.ts [hunk 3]
```diff
@@ -259,35 +315,34 @@ export class Stack<
...
+	 * @param file path to write the YAML file to
 	 */
-	static async writeTo(
-		stack: Stack,
-		file: string,
-		stringify: (obj: NitricStack) => string = YAML.stringify,
-	): Promise<void> {
-		const stackString = stringify(stack.asNitricStack(true)); //Turn object into yaml
+	static async writeTo(stack: Stack, file: string): Promise<void> {
+		const doc = insertComments(new YAML.Document(stack.asNitricStack(true)), stack.getComments());
```
Comment (tjholm) [inline packages/common/src/stack/stack.ts]: as above, can access private comments if its non-optional.

**Review by tjholm — CHANGES_REQUESTED** [review]

File: packages/common/src/stack/stack.ts (hunk 1 above)
Comment (jyecusch) [reply packages/common/src/stack/stack.ts]: Done

File: packages/common/src/stack/stack.ts (hunk 2 above)
Comment (jyecusch) [reply packages/common/src/stack/stack.ts]: Done

File: packages/common/src/stack/stack.ts (hunk 3 above)
Comment (jyecusch) [reply packages/common/src/stack/stack.ts]: Done

**Comment by jyecusch** [comment]
Resolved comments and added a basic test

**Review by tjholm — APPROVED** [review]

**Pull request merged**